BothamNetworking/BothamAPIClient.swift

```diff
@@ -104,15 +122,31 @@ public class BothamAPIClient {
         return interceptedRequest
     }
 
-    private func applyResponseInterceptors(response: HTTPResponse) -> HTTPResponse {
-        var interceptedResponse = response
-        responseInterceptors.forEach { interceptor in
-            interceptedResponse = interceptor.intercept(interceptedResponse)
-        }
-        BothamAPIClient.globalResponseInterceptors.forEach { interceptor in
-            interceptedResponse = interceptor.intercept(interceptedResponse)
-        }
-        return interceptedResponse
+    private func applyResponseInterceptors(response: HTTPResponse,
+        completion: (Result<HTTPResponse, BothamAPIClientError>) -> Void) {
+            let interceptors = responseInterceptors + BothamAPIClient.globalResponseInterceptors
+            applyResponseInterceptors(interceptors, response: response, completion: completion)
+    }
+
+    private func applyResponseInterceptors(interceptors: [BothamResponseInterceptor],
+        response: HTTPResponse,
+        completion: (Result<HTTPResponse, BothamAPIClientError>) -> Void) {
+            if interceptors.isEmpty {
+                completion(Result.Success(response))
+            } else if interceptors.count == 1 {
```

Comment:
I think we don't need the interceptors.count == 1
It's a short-circuit, but I don't think it's usefull.